src/adapters/InMemoryAdapter.js

```diff
@@ -18,20 +109,17 @@ class InMemoryAdapter extends Adapter {
       v => v.meta.type === modelName
     );
 
-    const values = typeValues.filter(v =>
-      Object.entries(query).reduce(
-        (result, [key, value]) => result && v[key] === value,
-        true
-      )
-    );
+    const values = typeValues.filter(v => runQueryFilter(v, query));
 
-    return values.reduce(
+    const test = values.reduce(
```

Comment:
I don't think `test` is a good variable name, I suggest either returning directly or calling the test results `results` or similar